week2/index.js

```diff
@@ -0,0 +1,213 @@
+    });
+}
+
+//  display the todo file once called
+function todoFileReader(fileName){
+    fs.exists(fileName, (exists) => {
+        //see if file exists to add the existing items to the array
+        if(exists){
+            fs.readFile(fileName, 'UTF-8', (err, data) => {
+                if (data === ""){
+                    console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+                    return;
+                } 
+                data = data.split('\r\n');
+                let text = data.reduce(((last, item, index) => last += `${index + 1}. ${item}\n`), ``);
+                console.log(text);
+            });
+        } else {
+            console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+        }
+    });
+}
+
+
+
+//check if there's already a file and push its contents into the todos array
+function checkFileContent(){
+    return new Promise(function(resolve, reject){
+        fs.exists('./todos.txt', (exists)=>{
+            if(!exists){
+                resolve(process.argv[3]);
+                return;                
+            }
+            fs.readFile('todos.txt', 'UTF-8', (err, data) => {
+                todos = data.trim().split('\r\n');
+                resolve(process.argv[3]);
+            });
+        });
+    });
+}
+
+//add the new item to the array
+function addTodoItem(todoItem){
+    if(todoItem !== ''){
+        if(todoItem.trim() !== ''){
+            todos.push(todoItem.trim());
+            todosWriter(todos);
+        }
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');        
+    }
+}
+
+// re-write existing file with new todos array 
+function todosWriter(itemsArr){
+    itemsArr.filter((a) => a !== '');
+    fs.writeFile('./todos.txt', itemsArr.join('\r\n'), 'UTF-8', (err) => {
+        if(err) console.log(err);
+    });
+}
+
+//remove item based on filled index
+function itemRemover(index){
+    //if index was actually filled in
+    if(index !== undefined){
+        index = parseInt(index) - 1;
+        fs.readFile('./todos.txt', 'UTF-8', (err, data) => {
+            data = data.split('\r\n');
+            //if that index actually existed in the todo list
+            if(data[index] !== undefined){
+                data.splice(index, 1);
+                todosWriter(data);
+                return;
+            }
+            console.log('You don\'t have that much items on your list!');
+        });
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');
+    }
+}
+
+//  empty the todos file
+function itemsResetter(){
+    fs.writeFile('./todos.txt', '', (err) => {
+        if (err) console.log(err);
+    });
+    console.log('Resetted the todo list');
+}
+
+
+//  modify an item based on filled index
+function update(index){
+    //check if an index was actually filled in
+    if(index !== undefined){
+        index = parseInt(index) - 1;
```

Comment:
What if index exists but isn't a valid int? What will happen here? (might be really bad!)

Reply:
Fixed! Good noticed haha I kept looking for these cases for a while but apparently not all of them are gone haha

Reply:
I don't know if you addressed my comment with this commit 😛 

Try running `parseInt( 'this is not a number' ) - 1`, and see if the result will cause problems when index is that value.